### ISSUE TITLE
Error when importing .svelte files in .ts files

### ISSUE THREAD

**Issue by vekunz** [import-js/eslint-plugin-import#2837]
Hi,

I am currently trying to migrate from `eslint-plugin-svelte3` to `eslint-plugin-svelte`. Most things are working, but when I import a `.svelte` file from a `.ts` file (for example in unit tests), I get two errors from `eslint-plugin-import` (at least I think the errors comes from this plugin).

```
Error while parsing C:\pathToRepo\src\routes\Counter.svelte
Line 10, column 19: Unexpected token :
`parseForESLint` from parser `svelte-eslint-parser` is invalid and will just be ignored
```

and

```
C:\pathToRepo\svelte-eslint-import-error\src\routes\Counter.test.ts
  2:21  warning  Parse errors in imported module './Counter.svelte': parser.parse is not a function (undefined:undefined)  import/no-named-as-default
```

I created a reproduction repo https://github.com/vekunz/svelte-eslint-import-error. Probably I still have a configuration issue somewhere, but after several hours of reading documentation and GitHub Issues, I could not find any solution to this.

**Comment by ljharb** [comment]
That's expected since .svelte isn't a standard thing. You'll need to have an eslint-import resolver for svelte files (presumably you already have the TS resolver).

**Comment by vekunz** [comment]
OK, I thought that the svelte-eslint-parser is enougth.

```js
	settings: {
		"import/resolver": {
			node: {},
			typescript: {
				alwaysTryTypes: false
			}
		},
		"import/parsers": {
			"svelte-eslint-parser": [".svelte"],
			"@typescript-eslint/parser": [".ts"],
			"espree": [".js"]
		}
	}
```

Do you know if there is any import resolver for svelte?

**Comment by ljharb** [comment]
Not that I'm aware of. Does https://github.com/import-js/eslint-plugin-import/issues/2407#issuecomment-1223394415 help?

**Comment by vekunz** [comment]
The recommendation there is to use eslint-plugin-svelte instead of the old eslint-plugin-svelte3, but I am already using the new plugin.

I remembered that eslint-plugin-import has an ignore-setting. With this option, ESLint at least doesn't show these errors, but the rest of ESLint seems to work.

```
  settings: {
    ...
    "import/ignore": [
      "\.svelte$"
    ]
  }
```

**Comment by ljharb** [comment]
Until there's a resolver, that might be your only option.

**Comment by fehnomenal** [comment]
I have the same problem. `svelte-plugin-eslint` for typescript ist configured like this:
```js
  overrides: [
    {
      files: ['*.svelte'],
      parser: 'svelte-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser',
      },
    },
  ],
```

To be able to import svelte files from js/ts we also need:
```js
  settings: {
    'import/parsers': {
      'svelte-eslint-parser': ['.svelte'],
    },
  },
```

But I guess here the `parserOptions.parser` for typescript is missing.

**Comment by ljharb** [comment]
eslint merges overrides; plugins don’t ever consider it.

**Comment by ljharb** [comment]
ah but we’re not getting the config for other files perhaps? I’ll be interested to review your pr.

**Comment by nonameolsson** [comment]
What is the current status of this? 🙂

**Comment by insilications** [comment]
Any solution?

**Comment by JounQin** [comment]
PR is always better than comments.

**Comment by vekunz** [comment]
We have this ESlint configuration, and this "works" (in the sense that we don't get errors).

```json
{
  "settings": {
    "import/resolver": {
      "node": {},
      "typescript": {
        "alwaysTryTypes": false
      }
    },
    "import/parsers": {
      "svelte-eslint-parser": [".svelte"],
      "@typescript-eslint/parser": [".ts"],
      "espree": [".js"]
    },
    "import/ignore": [
      "\\.svelte$"
    ]
  }
}
```

**Comment by niemyjski** [comment]
Svelte support is really lacking especially with import aliases.

**Comment by ljharb** [comment]
@niemyjski https://github.com/import-js/eslint-plugin-import/issues/2837#issuecomment-1646122243
